test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,366 @@
+        public void Append_WithTagHelperContent_MultipleAppends()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+            var text1 = "Hello";
+            var text2 = " World!";
+            var expected = text1 + text2;
+            tagHelperContent.Append(text1);
+            tagHelperContent.Append(text2);
+
+            // Act
+            copiedTagHelperContent.Append(tagHelperContent);
+
+            // Assert
+            Assert.Equal(expected, copiedTagHelperContent.GetContent());
+            Assert.Equal(new[] { text1, text2 }, copiedTagHelperContent.ToArray());
+        }
+
+        [Fact]
+        public void IsModified_TrueAfterSetContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent(string.Empty);
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        
+        [Fact]
+        public void IsModified_TrueAfterAppend()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.Append(string.Empty);
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        [Fact]
+        public void IsModified_TrueAfterClear()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData("\n")]
+        [InlineData("\t")]
+        [InlineData("\r")]
+        public void CanIdentifyWhiteSpace(string data)
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append(data);
+
+            // Assert
+            Assert.True(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void CanIdentifyWhiteSpace_WithoutIgnoringStrings()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append("Hello");
+
+            // Assert
+            Assert.False(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void IsEmpty_InitiallyFalse()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act & Assert
+            Assert.True(tagHelperContent.IsEmpty);
+        }
+
+        [Fact]
+        public void IsEmpty_FalseAfterSetEmptyContent()
```

Comment:
- False -> True here and in the next few tests (to match actual expectations)
- add (appropriately named) tests where `IsEmpty` ends up `false`